main.go

```diff
@@ -103,8 +105,16 @@ func buildExporter(opts GlobalFlags) *exporter.Exporter {
 
 	log.Debugf("Compatible mode: %v", opts.CompatibleMode)
 
-	if !strings.HasPrefix(opts.URI, "mongodb") {
-		log.Debugf("Prepending mongodb:// to the URI")
+	// IF user@pass not contained in uri AND custom user and pass supplied in arguments
+	// DO concat a new uri with user and pass arguments value
+	if !strings.Contains(opts.URI, "@") && opts.User != "" && opts.Password != "" {
```

Comment:
it should be two separate ifs, in case we have data source value like `username:password@host:port` without `mongodb://` prefix this if will be skipped.

Reply:
yes, and it changes current behavior and breaks backward compatibility, because before your changes we were prepending `mongodb://`

Reply:
yes, U are right, I will change it to achieve backward compatibility

Reply:
I have taken a look on changes carefully,
in this pr,
I add two new arguments `user` and `password`.

for backward compatibility, the two new  `user` and `password` arguments will be an empty string when not supplied, this `if` clause will not be triggered.
only when using new `user` and `password` argument is supplied and there is no user and pass in origin URI. this `if` will be triggered.

besides, if the user and password are both supplied in the origin URI and arguments(env or --user --pass), the exporter will take the URI one. (`if` not triggered )
the user, pass value in URI will have more priority than arguments

Reply:
Can we extract whole this logic and write a test so it will be clear what's wrong

the case which was working and won't work. `username:password@host:port` should be converted to `mongodb://username:password@host:port`

Reply:
have extracted logic, and write a test. https://github.com/percona/mongodb_exporter/pull/560/commits/1cdf1a6ae72f32b0ede18c1515122590abfadb02

but after I merged percona main to fork main, it stuck in `Some checks were not successful`

how can I fix this error, Should I revoke the merge operation?

Reply:
but the check is stucked by 

```
ok  	github.com/percona/mongodb_exporter/internal/tu	0.097s
?   	github.com/percona/mongodb_exporter/internal/util	[no test files]
FAIL
make: *** [Makefile:104: test-race] Error 1
```

This error seems unrelated to the code I've submitted.